context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { useEventCallback } from '@mui/material/utils';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import AddRounded from '@mui/icons-material/AddRounded';
+
+import SelectableTableProvider from 'js/shared-styles/tables/SelectableTableProvider';
+import { WorkspacesIcon } from 'js/shared-styles/icons';
+
+import { useOpenDialog } from '../../workspaces/WorkspacesDropdownMenu/WorkspacesDropdownMenu';
+import { useCreateWorkspaceForm } from '../../workspaces/NewWorkspaceDialog/useCreateWorkspaceForm';
+import NewWorkspaceDialog from '../../workspaces/NewWorkspaceDialog/NewWorkspaceDialog';
+import AddDatasetsFromSearchDialog from '../../workspaces/AddDatasetsFromSearchDialog/AddDatasetsFromSearchDialog';
+import { useAppContext, useFlaskDataContext } from '../../Contexts';
+import { useTrackEntityPageEvent } from '../useTrackEntityPageEvent';
+
+interface ProcessedDataWorkspaceMenuProps {
+  button: React.ReactNode;
+}
+
+function ProcessedDataWorkspaceMenu({ button }: ProcessedDataWorkspaceMenuProps) {
+  const {
+    entity: { mapped_data_access_level, hubmap_id, uuid, entity_type, status },
+  } = useFlaskDataContext();
+  const isDataset = entity_type === 'Dataset';
+  const { isWorkspacesUser } = useAppContext();
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
```

Comment:
Can we use `useRef` instead of `useState`.

Reply:
I actually have seen this pattern used in the MUI docs to handle open state, I.e. using this anchor element's presence in the state as a substitute for an "open" boolean. I don't hate it as long as that logic remains intact (I.e open = Boolean(anchorEl))